toolkit/tools/pkg/imagecustomizerlib/customizeboot.go

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/safechroot"
+)
+
+var (
+	linuxCommandLineRegex = regexp.MustCompile(`\tlinux .* (\$kernelopts)`)
+)
+
+func handleKernelCommandLine(commandLineAdd string, imageChroot *safechroot.Chroot) error {
+	var err error
+
+	if commandLineAdd == "" {
+		// Nothing to do.
+		return nil
+	}
+
+	grub2ConfigFilePath := filepath.Join(imageChroot.RootDir(), "/boot/grub2/grub.cfg")
```

Comment:
@camelron has been working on using grub2 mkconfig to dynamically generate `grub.cfg`. If a user is using this new flow `/boot/grab2/grub.cfg` will get overwritten. We would need to update both the template input and this file (or just trigger mkconfig again to update).

This is mostly for M3, but we should probably handle it here so we don't have any bad surprises in the future. I'll get him to comment on what we should check for.

Reply:
These are the two files of interest: https://github.com/microsoft/CBL-Mariner/blob/eebc0beedc34649f3f10bb9e1763afbef531080c/toolkit/tools/imagegen/installutils/installutils.go#L1088

Reply:
per offline, change will come in future PR.

Reply:
This isn't grub-enlightened; changes to /boot/grub2/grub.cfg get stomped next time grub2-mkconfig is called. 

You could drop all imagecustomizer settings in an /etc/default/grub.d/10_customizer_cfg.cfg 

See example: https://github.com/microsoft/CBL-Mariner/blob/2.0/SPECS/kernel-mshv/50_mariner_mshv.cfg#L4C2-L4C2 

This will cause any additional cmdline params to stream into the generated /boot/grub2/grub.cfg every time.